Fix to show commitPrefixes in commit message with a new blank commit

longlhh90 opened the pull request:
- **PR Description**

Yesterday, when I updated lazygit to the newest version, my commitPrefixes is not working anymore. Actually, it's not working when I start lazygit and start a new commit with blank message. If I commit it as WIP or add some text and cancel and recommit again prefix will show.

It was working perfectly fine in old version (I couldn't remember which one).

The fix is: 
  - remove the check message != ""
  - check if message doesn't contain prefix => add prefix in front of message
  - add tests

I commented on this issue #2215 however I don't know if they have same issue as me.

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

tsjk commented:
Great. I opened an issue about this just now.

jesseduffield commented:
@longlhh90 could you add an integration test for this which fails under the old code and succeeds under the new code?

longlhh90 commented:
> @longlhh90 could you add an integration test for this which fails under the old code and succeeds under the new code?

Hi @jesseduffield, I'm on it

longlhh90 commented:
> @longlhh90 that's likely just a flakey test that's failing, so I wouldn't worry about it. But yeah we want to add a new integration test for this PR

I added some tests to verify logic changes. I also changed a bit when there is prefix defined. Currently, whenever user cancel commit popup, commit will be overridden by prefix. For example, `[Prefix] type something` => cancel => commit again => `[Prefix] `.

So, I added a check to add prefix to commit message only when it isn't there.

jesseduffield commented:
@longlhh90 I've pushed a change to the logic: It turns out that the issue was simply that the `!=` should have been a `==`. When it comes to WIP commits, we don't want to use the prefix because we have logic checking that 'WIP' is the prefix of the commit if we're to skip hooks.

We can improve this logic to make it so that we set an internal flag that says we want to skip hooks, but that can come later. Let's merge this PR to fix the original problem.

jesseduffield commented:
Thanks for making this @longlhh90 ! And good work on the integration tests

longlhh90 commented:
> @longlhh90 I've pushed a change to the logic: It turns out that the issue was simply that the `!=` should have been a `==`. When it comes to WIP commits, we don't want to use the prefix because we have logic checking that 'WIP' is the prefix of the commit if we're to skip hooks.
> 
> We can improve this logic to make it so that we set an internal flag that says we want to skip hooks, but that can come later. Let's merge this PR to fix the original problem.

sounds good!
